tests/lib/c_lib/src/test_sqrt.c

```diff
@@ -95,31 +97,58 @@ static int isnanf(float x)
 ZTEST(test_c_lib, test_sqrtf)
 {
 int i;
-float	exponent;
-float	resf;
+float	exponent, resf, square, root_squared;
 double	error;
+uint32_t max_error = 0;
+int32_t ierror;
+int32_t *p_square = (int32_t *)&square;
+int32_t *p_root_squared = (int32_t *)&root_squared;
 
 
+	/*
+	 * sqrtf not supported without FPU except on native POSIX.
+	 */
+        if ( !(IS_ENABLED(CONFIG_FPU)
```

Comment:
This is not correct -- sqrtf should be available on all platforms as those without an FPU will use soft float. And testing to make sure sqrtf works with soft float is pretty important as soft float implementations are not always complete.

Reply:
I am having problems with the twister runs, I get this error message:

START - test_sqrt
E: Floating point unit device not available

for these tests:
west twister -p qemu_x86 -s tests/lib/c_lib/libraries.libc.picolibc
west twister -p qemu_leon3 -s tests/lib/c_lib/libraries.libc.picolibc

Testing for CONFIG_FPU  makes the runs pass. If this is not the right thing to test, then what should I be testing?